hw/ip/pattgen/data/pattgen_testplan.hjson

```diff
@@ -0,0 +1,99 @@
+
+            Stimulus:
+              - Program the configuration registers of the output channels
+              - Randomly activate the output channels
+              - Re-program the configuration registers of the output channels once 
+                completion interrupts are asserted
+
+            Checking:
+              - Check divided clock rate for the active channels
+                matching with the values of pre-divider registers
+              - Check generated pattern matching on the active channels
+                matching with the values of pattern data registers
+              - Check completion interrupts are asserted once a pattern
+                is completely generated on the active channels
+            '''
+      milestone: V1
+      tests: ["pattgen_sanity"]
+    }
+    {
+      name: intr_perf
+      desc: '''
+            Checking ip operation at min/max bandwith
+
+            Stimulus:
+              - Program the pre-divider registers to highest/lowest values (lowest/highest data rate)
+              - Program the pattern data registers, the pattern length per output, and 
+                repeat counter registers to smallest values
+              - Clear interrupts quickly
+
+            Checking:
+              - Ensure patterns are correctly generated
+              - Ensure interrupts are robust asserted and cleared (e.g. at the highest data rate)
+            '''
+      milestone: V2
+      tests: ["pattgen_perf"]
+    }   
+    {
+      name: error
+      desc: '''
+            Reset then re-start the output channel on the fly.
+
+            Stimulus:
+              - Programm the configuration registers of the output channels
+              - Randomly reset the output channel in progress
+              - Re-program the configuration registers of the output channels
+
+            Checking:
+              - Ensure patterns are dropped when reset
+              - Ensure the output channels get back normal after reset
+            '''
+      milestone: V2
+      tests: ["pattgen_error"]
+    }
+    {
+      name: stress_all
+      desc: '''
+            Combine above sequences in one test to run sequentially,
+
+            Stimulus:
+              - Start sequences and randomly add reset between each sequence
+
+            Checking:
+              - All sequences should finished and checked by the scoreboard
+      '''
+      milestone: V2
+      tests: ["pattgen_stress_all"]
+    }
+    {
+      name: stress_all_with_reset
+      desc: '''
+            Pull reset at random times, ensure dut are reset and recovered/correctly and
+            there is no residual data left in the registers.
+
+            Stimulus:
+              - Randomly reset DUT
+              - Restart pattern generation
+
+            Checking:
+              - Ensure registers are properly reset
+            ''' 
+      milestone: V2
+      tests: ["pattgen_stress_all_with_rand_reset"]
```

Comment:
Do you plan to use standard `stress_all_with_rand_reset` , if so, maybe you can directly import from here: https://github.com/lowRISC/opentitan/blob/master/hw/dv/tools/testplans/stress_all_with_reset_testplan.hjson.

Reply:
It should be standard `stress_all_with_rand_reset` but I think we still need to define that in the testplan even it requires zero-cost of development at IP level

Reply:
Thanks for confirming, if this is the standard stress_all_with_rand_reset, you can just import the testplan like the CSR_test_suites in line 6-8 :)

Reply:
Got your point, remove  the description for `stress_all_with_rand_reset` by importing `hw/dv/tools/testplans/stress_all_with_reset_testplan.hjson`